### PR TITLE
753: add checkbox for static qr code verification

### PR DESCRIPTION
**Pull request by f1sh1918**
resolves https://github.com/digitalfabrik/entitlementcard/issues/753

### CODE REVIEW

**Comment by maxammann** [comment]
@f1sh1918 I rebased this onto main!

File: frontend/lib/identification/verification_workflow/dialogs/positive_verification_result_dialog.dart [hunk 1]
```diff
@@ -28,19 +53,43 @@ class PositiveVerificationResultDialog extends StatelessWidget {
...
+                  region: region != null ? Region(region.prefix, region.name) : null,
+                ),
+              ),
+              if (widget.hasStaticVerificationCode)
+                Flexible(
+                  child: Padding(
+                    padding: const EdgeInsets.only(top: 16.0),
+                    child: CheckboxListTile(
+                      title: const Text('Ich habe die Daten mit einem amtlichen Lichtbildausweis abgeglichen.'),
+                      checkColor: Theme.of(context).primaryColor,
```
Comment (michael-markl) [inline frontend/lib/identification/verification_workflow/dialogs/positive_verification_result_dialog.dart:76]: Please don't adjust the checkbox color here (it looks weird on light theme)
You can adjust the checkboxTheme for the darkmode in themes.dart
Under `theme.copyWith(...)` add something like:

    checkboxTheme: CheckboxThemeData(
      checkColor: const MaterialStatePropertyAll(Colors.white),
      fillColor: MaterialStatePropertyAll(primaryColor),
    ),

File: frontend/lib/identification/verification_workflow/dialogs/positive_verification_result_dialog.dart [hunk 2]
```diff
@@ -28,19 +53,43 @@ class PositiveVerificationResultDialog extends StatelessWidget {
...
         final region = result.isConcrete && data != null ? regionsQuery.parse(data).regionsByIdInProject[0] : null;
+        final bool isUncheckedStaticQrCode = !isChecked && widget.hasStaticVerificationCode;
         return InfoDialog(
-          title: localization.positiveVerificationDialogTitle,
-          icon: Icons.verified_user,
-          iconColor: Colors.green,
-          child: IdCard(
-            cardInfo: cardInfo,
-            region: region != null ? Region(region.prefix, region.name) : null,
+          title: isUncheckedStaticQrCode ? "Prüfung notwendig" : localization.positiveVerificationDialogTitle,
```
Comment (michael-markl) [inline frontend/lib/identification/verification_workflow/dialogs/positive_verification_result_dialog.dart:58]:
```suggestion
          title: isUncheckedStaticQrCode ? "Prüfung nötig" : localization.positiveVerificationDialogTitle,
```
(Increasing the chance that it fits in one line)

**Review by michael-markl — APPROVED** [review]
Looks good, tested on Android 👍🏽

File: frontend/lib/identification/verification_workflow/dialogs/positive_verification_result_dialog.dart [hunk 3]
```diff
@@ -28,19 +53,43 @@ class PositiveVerificationResultDialog extends StatelessWidget {
...
+                  child: Padding(
+                    padding: const EdgeInsets.only(top: 16.0),
+                    child: CheckboxListTile(
+                      title: const Text('Ich habe die Daten mit einem amtlichen Lichtbildausweis abgeglichen.'),
+                      checkColor: Theme.of(context).primaryColor,
+                      controlAffinity: ListTileControlAffinity.leading,
+                      value: isChecked,
+                      onChanged: (bool? value) {
+                        setState(() {
+                          isChecked = value!;
```
Comment (michael-markl) [inline frontend/lib/identification/verification_workflow/dialogs/positive_verification_result_dialog.dart:81]:
```suggestion
                          isChecked = value ?? false;
```

File: frontend/lib/identification/verification_workflow/dialogs/positive_verification_result_dialog.dart [hunk 4]
```diff
@@ -28,19 +53,43 @@ class PositiveVerificationResultDialog extends StatelessWidget {
...
+                        });
+                      },
+                    ),
+                  ),
+                )
+              else
+                Container()
+            ],
```
[the remark this answers is not in the document]
Comment (f1sh1918) [reply frontend/lib/identification/verification_workflow/dialogs/positive_verification_result_dialog.dart]: component won't rendered without flexible but removed the `Container()`

File: frontend/lib/identification/verification_workflow/dialogs/positive_verification_result_dialog.dart [hunk 5]
```diff
@@ -28,19 +53,43 @@ class PositiveVerificationResultDialog extends StatelessWidget {
...
+        final bool isUncheckedStaticQrCode = !isChecked && widget.hasStaticVerificationCode;
         return InfoDialog(
-          title: localization.positiveVerificationDialogTitle,
-          icon: Icons.verified_user,
-          iconColor: Colors.green,
-          child: IdCard(
-            cardInfo: cardInfo,
-            region: region != null ? Region(region.prefix, region.name) : null,
+          title: isUncheckedStaticQrCode ? "Prüfung notwendig" : localization.positiveVerificationDialogTitle,
+          icon: isUncheckedStaticQrCode ? Icons.info : Icons.verified_user,
```
[the remark this answers is not in the document]
Comment (f1sh1918) [reply frontend/lib/identification/verification_workflow/dialogs/positive_verification_result_dialog.dart:59]: `Icons.report` looks fine

**Pull request merged**